notification/slack/channel.go

```diff
@@ -150,8 +170,9 @@ func (s *ChannelSender) loadChannel(ctx context.Context, channelID string) (*Cha
 	}
 
 	return &Channel{
-		ID:   resData.Channel.ID,
-		Name: "#" + resData.Channel.Name,
+		ID:     resData.Channel.ID,
+		Name:   "#" + resData.Channel.Name,
+		TeamID: s.teamID,
```

Comment:
We can't access `s.teamID` outside of the mutex, otherwise, there's the potential for concurrent read & write access. Assigning a copy to a variable before `Unlock` can work around this.

```go
teamID := s.teamID
s.teamMx.Unlock()

return &Channel{
	TeamID: teamID,
}
```